bot/decorators.py

```diff
@@ -3,15 +3,21 @@
 from discord.ext.commands import Context
 
 
+def invoke_with_prefix(prefix: str) -> commands.check:
+    '''Allow a command to only be invoked with one prefix
+    :str: prefix - The prefix to be triggered with
+    '''
+    async def predicate(ctx: Context):
+        return ctx.prefix == prefix
+    return commands.check(predicate)
+
+
 def with_role(*role_ids: int):
     async def predicate(ctx: Context):
         if not ctx.guild:  # Return False in a DM
             return False
 
-        for role in ctx.author.roles:
-            if role.id in role_ids:
-                return True
-        return False
+        return any((role.id in role_ids) for role in ctx.author.roles)
```

Comment:
The logic is the exact same, but the readability is lowered slightly. I'd suggest keeping it as it was before, but that's open to discussion.

Reply:
Sure, I don't mind.

Reply:
![quotes](https://i.imgur.com/ohCQeoR.png)

Just going to leave this here...

Reply:
readability is worse. the original logic is basically an english sentence. for these if this then that. yours reads if any of these are true and by these I mean that the id of the role is in this list and by role I mean the elements that are generated by this expression - it's all backwards. sequence is a critical part of how the brain parses information.